src/commands/force/mdapi/beta/deploy/report.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { Messages } from '@salesforce/core';
+import { flags, FlagsConfig } from '@salesforce/command';
+import { Duration, env } from '@salesforce/kit';
+import { RequestStatus } from '@salesforce/source-deploy-retrieve';
+
+import { MdDeployResult, MdDeployResultFormatter } from '../../../../../formatters/mdDeployResultFormatter';
+import { DeployCommand } from '../../../../../deployCommand';
+import { ProgressFormatter } from '../../../../../formatters/progressFormatter';
+import { DeployProgressBarFormatter } from '../../../../../formatters/deployProgressBarFormatter';
+import { DeployProgressStatusFormatter } from '../../../../../formatters/deployProgressStatusFormatter';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-source', 'md.deployreport');
+
+export class Report extends DeployCommand {
+  public static readonly description = messages.getMessage('description');
+  public static readonly requiresUsername = true;
+
+  public static readonly flagsConfig: FlagsConfig = {
+    wait: flags.minutes({
+      char: 'w',
+      default: Duration.minutes(DeployCommand.DEFAULT_WAIT_MINUTES),
+      min: Duration.minutes(1),
+      description: messages.getMessage('flags.wait', [DeployCommand.DEFAULT_WAIT_MINUTES]),
+      longDescription: messages.getMessage('flagsLong.wait', [DeployCommand.DEFAULT_WAIT_MINUTES]),
+    }),
+    jobid: flags.id({
+      char: 'i',
+      description: messages.getMessage('flags.jobId'),
+      longDescription: messages.getMessage('flagsLong.jobId'),
+    }),
+    verbose: flags.builtin({
+      description: messages.getMessage('flags.verbose'),
+      longDescription: messages.getMessage('flagsLong.verbose'),
+    }),
+  };
+
+  public async run(): Promise<MdDeployResult> {
+    await this.deploy();
+    this.resolveSuccess();
+    return this.formatResult();
+  }
+
+  protected async deploy(): Promise<void> {
```

Comment:
nit: `deploy` isn't an accurate name for this.  Maybe `report` or `checkStatus` or something.

Reply:
will make it match yours from `source`